### PR TITLE
SDIT-2176 Move restricted patients to domain partial update

### PR DESCRIPTION
**Pull request by steverendell**
RPs now work the same as incentives in the RED index

### CODE REVIEW

File: hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/PrisonerSynchroniserServiceTest.kt [hunk 1]
```diff
@@ -269,69 +272,6 @@ internal class PrisonerSynchroniserServiceTest {
...
-      )
-      whenever(restrictedPatientService.getRestrictedPatient(any())).thenThrow(RuntimeException("not today thank you"))
-      whenever(prisonerRepository.getSummary(any(), eq(RED))).thenReturn(prisonerDocumentSummary)
-
-      assertThatThrownBy { service.reindexUpdate(outsidePrisoner, "event") }
-        .hasMessage("not today thank you")
-
-      verify(prisonerRepository, times(1)).updatePrisoner(isA(), isA(), isA(), isA())
-    }
-
```
Comment (steverendell) [inline hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/services/PrisonerSynchroniserServiceTest.kt:334]: moved these tests

File: common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/translator.kt [hunk 2]
```diff
@@ -8,6 +8,19 @@ import uk.gov.justice.digital.hmpps.prisonersearch.common.nomis.OffenceHistoryDe
 import uk.gov.justice.digital.hmpps.prisonersearch.common.nomis.OffenderBooking
 import uk.gov.justice.digital.hmpps.prisonersearch.common.nomis.OffenderIdentifier
 import uk.gov.justice.digital.hmpps.prisonersearch.common.nomis.Telephone
+import kotlin.collections.filter
```
Comment (petergphillips) [inline common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/translator.kt:11]: lots of unnecessary imports?

**Review by andymarke — DISMISSED** [review]
LGTM

**Review by petergphillips — APPROVED** [review]
LGTM

File: common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/translator.kt (hunk 2 above)
Comment (steverendell) [reply common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/translator.kt:11]: Thats weird, my intellij doesnt flag them as unused. But i thought the linter objected to unused imports. Will remove

File: common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/translator.kt [hunk 3]
```diff
@@ -166,6 +159,21 @@ fun IncentiveLevel?.toCurrentIncentive(): CurrentIncentive? = this?.let {
...
 }
 
+fun Prisoner.setRestrictedPatient(
+  rp: RestrictedPatient?,
+  ob: OffenderBooking,
+) {
+  this.restrictedPatient = rp != null
+  this.locationDescription = rp
+    ?.let { "${ob.locationDescription} - discharged to ${it.dischargedHospital?.description}" }
+    ?: ob.locationDescription
```
Comment (petergphillips) [inline common/src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/common/model/translator.kt:169]: this is a bit odd - if they are not a restricted patient then this is still set even though the method is called `setRestrictedPatient`

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**